client_plugin/drivers/shared/kvstore/etcdops/etcdops.go

```diff
@@ -338,12 +346,44 @@ func (e *EtcdKVS) etcdEventHandler(ev *etcdClient.Event) {
 			return
 		}
 
-		if fn(volName) {
-			// server start/stop succeed, set desired state
-			if e.CompareAndPut(kvstore.VolPrefixState+volName,
+		port, servName, succeeded := fn(volName)
+		if succeeded {
+			// Update volume metadata to reflect
+			// port number and file service name
+			var entries []kvstore.KvPair
+			var volRecord sharedVolConnectivityData
+			volRecord.Port = port
+			volRecord.ServiceName = servName
+			byteRecord, err := json.Marshal(volRecord)
+			if err != nil {
+				log.Warningf("Failed to update port and file service name: %v",
+					err)
+				e.CompareAndPut(kvstore.VolPrefixState+volName,
+					string(interimState),
+					string(kvstore.VolStateError))
+				return
+			}
+			entries = append(entries, kvstore.KvPair{
+				Key:   kvstore.VolPrefixInfo + volName,
+				Value: string(byteRecord)})
```

Comment:
Question: the sharedVolConnectivityData only has Port and ServiceName field. So the byteRecord will only contain these two fields. If we writeMetaData with this byteRecord string, won't this operation overwrite other existing fields for the value of Info key? such as the username/password, clients...